music_assistant/server/providers/plex/__init__.py

```diff
@@ -556,10 +559,13 @@ async def get_album_tracks(self, prov_album_id: str) -> list[AlbumTrack]:
         """Get album tracks for given album id."""
         plex_album: PlexAlbum = await self._get_data(prov_album_id, PlexAlbum)
         tracks = []
-        for plex_track in await self._run_async(plex_album.tracks):
+        for idx, plex_track in enumerate(await self._run_async(plex_album.tracks), 1):
             track = await self._parse_track(
                 plex_track,
-                {"disc_number": plex_track.parentIndex, "track_number": plex_track.trackNumber},
+                {
+                    "disc_number": plex_track.parentIndex,
+                    "track_number": idx,
```

Comment:
Is this on purpose ? Now you are overwriting any value plex had already
maybe change to "track_number": plex_track.trackNumber or idx